File: mobile-app/playconnect_mobile/lib/src/services/api/api_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiService {
  static const String baseUrl = 'http://localhost:3003';
  static const String aiBaseUrl = 'http://localhost:3009';
  static const String fileBaseUrl = 'http://localhost:3006';
  static const String videoBaseUrl = 'http://localhost:3008';

  static final ApiService _instance = ApiService._internal();
  factory ApiService() => _instance;
  ApiService._internal();

  final http.Client _client = http.Client();

  // Player Service Methods
  Future<List<dynamic>> getPlayers({Map<String, String>? filters}) async {
    try {
      final queryParams = filters != null ? '?${Uri(queryParameters: filters).query}' : '';
      final response = await _client.get(Uri.parse('$baseUrl/players$queryParams'));
      
      if (response.statusCode == 200) {
        final data = json.decode(response.body);
        return data['players'] ?? [];
      } else {
        throw Exception('Failed to load players: ${response.statusCode}');
      }
    } catch (e) {
      throw Exception('Failed to load players: $e');
    }
  }

  Future<dynamic> getPlayer(String id) async {
    try {
      final response = await _client.get(Uri.parse('$baseUrl/players/$id'));
      
      if (response.statusCode == 200) {
        return json.decode(response.body);
      } else {
        throw Exception('Failed to load player: ${response.statusCode}');
      }
    } catch (e) {
      throw Exception('Failed to load player: $e');
    }
  }

  // AI Service Methods
  Future<List<dynamic>> discoverTalent(Map<String, dynamic> criteria) async {
    try {
      final response = await _client.post(
        Uri.parse('$aiBaseUrl/api/ai/talent-discovery'),
        headers: {'Content-Type': 'application/json'},
        body: json.encode(criteria),
      );
      
      if (response.statusCode == 200) {
        final data = json.decode(response.body);
        return data['data']['players'] ?? [];
      } else {
        throw Exception('Failed to discover talent: ${response.statusCode}');
      }
    } catch (e) {
      throw Exception('Failed to discover talent: $e');
    }
  }

  // File Service Methods
  Future<dynamic> uploadVideo(List<int> videoBytes, String filename) async {
    try {
      final request = http.MultipartRequest(
        'POST',
        Uri.parse('$fileBaseUrl/api/videos/upload'),
      );
      
      request.files.add(http.MultipartFile.fromBytes(
        'video',
        videoBytes,
        filename: filename,
      ));
      
      final response = await _client.send(request);
      final responseBody = await response.stream.bytesToString();
      
      if (response.statusCode == 200) {
        return json.decode(responseBody);
      } else {
        throw Exception('Failed to upload video: ${response.statusCode}');
      }
    } catch (e) {
      throw Exception('Failed to upload video: $e');
    }
  }

  // Video Service Methods
  Future<dynamic> analyzeVideo(String videoId) async {
    try {
      final response = await _client.get(
        Uri.parse('$videoBaseUrl/api/videos/analysis/$videoId'),
      );
      
      if (response.statusCode == 200) {
        return json.decode(response.body);
      } else {
        throw Exception('Failed to analyze video: ${response.statusCode}');
      }
    } catch (e) {
      throw Exception('Failed to analyze video: $e');
    }
  }
}

File: frontend/federation-dashboard/src/services/apiService.js
const API_BASE_URL = process.env.REACT_APP_API_URL || 'http://localhost:3000';

class ApiService {
  constructor() {
    this.baseURL = API_BASE_URL;
  }

  async request(endpoint, options = {}) {
    const url = `${this.baseURL}${endpoint}`;
    const config = {
      headers: {
        'Content-Type': 'application/json',
        ...options.headers,
      },
      ...options,
    };

    // Add auth token if available
    const token = localStorage.getItem('authToken');
    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
    }

    try {
      const response = await fetch(url, config);
      
      if (!response.ok) {
        const errorData = await response.json().catch(() => ({}));
        throw new Error(errorData.error || `HTTP error! status: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      console.error('API request failed:', error);
      throw error;
    }
  }

  // Player Management APIs - Connect to existing player service
  async getPlayers(filters = {}) {
    const queryParams = new URLSearchParams(filters).toString();
    return this.request(`/api/players?${queryParams}`);
  }

  async getPlayer(id) {
    return this.request(`/api/players/${id}`);
  }

  async createPlayer(playerData) {
    return this.request('/api/players', {
      method: 'POST',
      body: JSON.stringify(playerData),
    });
  }

  async updatePlayer(id, playerData) {
    return this.request(`/api/players/${id}`, {
      method: 'PUT',
      body: JSON.stringify(playerData),
    });
  }

  async deletePlayer(id) {
    return this.request(`/api/players/${id}`, {
      method: 'DELETE',
    });
  }

  // Video Management APIs
  async uploadVideo(videoFile, playerId) {
    const formData = new FormData();
    formData.append('video', videoFile);
    formData.append('playerId', playerId);

    return this.request('/api/videos/upload', {
      method: 'POST',
      headers: {
        // Don't set Content-Type for FormData, browser will set it with boundary
      },
      body: formData,
    });
  }

  async getPlayerVideos(playerId) {
    return this.request(`/api/players/${playerId}/videos`);
  }

  async deleteVideo(videoId) {
    return this.request(`/api/videos/${videoId}`, {
      method: 'DELETE',
    });
  }

  // User Management APIs
  async getUsers() {
    return this.request('/api/users');
  }

  async updateUserRole(userId, role) {
    return this.request(`/api/users/${userId}/role`, {
      method: 'PUT',
      body: JSON.stringify({ role }),
    });
  }

  // Federation Management APIs
  async getFederations() {
    return this.request('/api/federations');
  }

  async getFederationPlayers(federationId) {
    return this.request(`/api/federations/${federationId}/players`);
  }

  // Search APIs
  async searchPlayers(query, filters = {}) {
    return this.request('/api/search/players', {
      method: 'POST',
      body: JSON.stringify({ query, filters }),
    });
  }

  // Analytics APIs
  async getPlayerAnalytics(playerId) {
    return this.request(`/api/analytics/players/${playerId}`);
  }

  async getFederationAnalytics(federationId) {
    return this.request(`/api/analytics/federations/${federationId}`);
  }
}

export default new ApiService();
